static/src/js/actions/facets.js

```diff
@@ -0,0 +1,70 @@
+import { camelCase } from 'lodash'
+
+import {
+  ADD_SELECTED_FEATURE_FACET,
+  REMOVE_SELECTED_FEATURE_FACET,
+  ADD_SELECTED_CMR_FACET,
+  REMOVE_SELECTED_CMR_FACET
+} from '../constants/actionTypes'
+
+import actions from './index'
+
+
+/**
+ * Sets the desired feature facet and performs a collections request.
+ * @param {object} e - An event object triggered by the onChange event on the selected facet.
+ * @param {object} facetInfo - An object containing information about the selected facet.
+ * @param {function} dispatch - A dispatch function provided by redux.
+ */
+
+export const changeFeatureFacet = (e, facetInfo) => (dispatch) => {
+  if (e.target.checked) {
+    dispatch({
+      type: ADD_SELECTED_FEATURE_FACET,
+      payload: { [camelCase(facetInfo.title)]: true }
+    })
+  } else {
+    dispatch({
+      type: REMOVE_SELECTED_FEATURE_FACET,
+      payload: { [camelCase(facetInfo.title)]: false }
+    })
+  }
+  dispatch(actions.getCollections())
+}
+
+
+/**
+ * Sets the desired facet and performs a collections request.
+ * @param {object} e - An event object triggered by the onChange event on the selected facet.
+ * @param {object} newParams - An object containing the params from the apply/remove link
+ * @param {function} dispatch - A dispatch function provided by redux.
+ */
+
+export const changeCmrFacet = (e, newParams) => (dispatch) => {
+  if (e.target.checked) {
+    dispatch({
+      type: ADD_SELECTED_CMR_FACET,
+      payload: {
+        science_keywords_h: newParams.science_keywords_h ? newParams.science_keywords_h : undefined,
```

Comment:
Using the ternary here wont these parameters always be sent to CMR? It may be more verbose but I think it would be best to prevent sending undefined params to CMR.

Reply:
Any undefined value will not be set in the Redux store. This is really making sure that any new facet request overwrites missing values in the store. Otherwise, they would stick around once we spread out the values in the reducer.